Implemented functional Destiny 2 api calls

sylphia-nox opened the pull request:
Currently just a draft.

sylphia-nox committed:
Beginning work with APIs

Initial work, ~character_ids <steam name: str> is fully functional.

michaelScarfi commented:
Initial work, ~character_ids <steam name: str> is fully functional.

sylphia-nox committed:
New ~next_level command

Lots of changes but new command to show current level and distance to next level

michaelScarfi commented:
Initial work is complete, however, I would like to give the new branch some testing time before merging.

sylphia-nox committed:
Fixed issue with API calls for non-steam origin accounts

With cross-save, some users are not under Steam as their account type, made the code able to handle this and to still work properly.

sylphia-nox committed:
Removed character_ids command

Removed uneeded command and renamed ~next_level to ~power
